File: 二叉树/100.cpp
#include<queue>
using namespace std;
//Definition for a binary tree node.
struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class DFS
{
public:
    bool isSameTree(TreeNode *p, TreeNode *q)
    {
        if ((p && !q) || (!p && q))
            return false;
        else if (!p && !q)
            return true;
        else
        {
            if (p->val == q->val)
                return isSameTree(p->left, q->left) && isSameTree(p->right, q->right);
            else
                return false;
        }
    }
};

class BFS
{
public:
    bool isSameTree(TreeNode*p,TreeNode*q)
    {
        queue<TreeNode*> NodeList;
        NodeList.push(p);
        NodeList.push(q);
        while(!NodeList.empty())
        {
            p = NodeList.front();
            NodeList.pop();
            q = NodeList.front();
            NodeList.pop();
            if ((p && !q) || (!p && q))
                return false;
            if(!p&&!q)
                continue;
            if(p->val!=q->val)
                return false;
            NodeList.push(p->left);
            NodeList.push(q->left);
            NodeList.push(p->right);
            NodeList.push(q->right);
        }
        return true;
    }
};

File: 二叉树/100.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class DFS:
    def isSameTree(p: TreeNode, q: TreeNode) -> bool:
        if(p and not q) or (not p and q):
            return False
        elif (not p and not q):
            return True
        elif (p.val != q.val):
            return False
        else:
            return self.isSameTree(p.left, q.left) and self.isSameTree(p.right, q.right)


class BFS:
    def isSameTree(p: TreeNode, q: TreeNode) -> bool:
        NodeList = []
        NodeList.append(p)
        NodeList.append(q)
        while(NodeList):
            p = NodeList.pop(0)
            q = NodeList.pop(0)
            if(p and not q) or (not p and q):
                return False
            elif (p.val != q.val):
                return False
            elif (not p and not q):
                continue
            else:
                NodeList.append(p.left)
                NodeList.append(q.left)
                NodeList.append(p.right)
                NodeList.append(q.right)
        return True